### PR TITLE
Stream consistency bug fixes and test additions

### PR DESCRIPTION
**Pull request by vaibhav-yb**

### CODE REVIEW

**Review by vrajat — APPROVED** [review]

**Review by vrajat — APPROVED** [review]
LGTM up to this point.

File: src/main/java/io/debezium/connector/yugabytedb/consistent/Merger.java [hunk 1]
```diff
@@ -4,22 +4,29 @@
...
 
+/**
+ * @author Rajat Venkatesh
+ */
 public class Merger {
     private static final Logger LOGGER = LoggerFactory.getLogger(Merger.class);
-    private final PriorityQueue<Message> queue = new PriorityQueue<>();
-    private final Map<String, List<Message>> mergeSlots = new HashMap<>();
-    private final Map<String, Long> tabletSafeTime = new HashMap<>();
+    private final PriorityBlockingQueue<Message> queue = new PriorityBlockingQueue<>();
```
Comment (vrajat) [inline src/main/java/io/debezium/connector/yugabytedb/consistent/Merger.java:18]: Can you move the data structures back to non-concurrent ones? There is no point in taking locks at function and data structure level.

**Review by vrajat — CHANGES_REQUESTED** [review]

File: src/main/java/io/debezium/connector/yugabytedb/consistent/Merger.java [hunk 2]
```diff
@@ -31,32 +35,51 @@ public void addMessage(Message message) {
...
         return peeked;
     }
 
-    public Message poll() {
+    public synchronized Optional<Message> poll() {
+        if (queue.isEmpty()) {
+            return Optional.empty();
+        }
+
         Message message = Objects.requireNonNull(queue.poll());
```
Comment (vrajat) [inline src/main/java/io/debezium/connector/yugabytedb/consistent/Merger.java:72]: `poll` removes the message. However, the next if condition maybe false and might lose the message. Instead the code should `peek` instead. 

My suggestion is turn `Merger::peek` to private and then use it in `Merger::poll`. So the code structure is as follows:

```
public synchronised Optional<Message> poll() {
    Message message = this.peek();
    // No valid message to emit
    if (message.isEmpty())
        return message;
        
    // Remove message from queue and mergeSlots
    queue.poll();
    mergeSlots.get(...).removeIf(...);
    return message;    
}
```

**Review by vrajat — CHANGES_REQUESTED** [review]

File: src/main/java/io/debezium/connector/yugabytedb/consistent/Merger.java (hunk 2 above)
Comment (vaibhav-yb) [reply src/main/java/io/debezium/connector/yugabytedb/consistent/Merger.java]: Addressed.

**Review by vrajat — APPROVED** [review]
One minor comment. A lot of the log messages are INFO. Do you want to move it to DEBUG?

**Comment by vaibhav-yb** [comment]
> One minor comment. A lot of the log messages are INFO. Do you want to move it to DEBUG?

@vrajat Wanted to keep the info logs for now as they will help in further debugging in our tests as of now. I will make them as DEBUG once a PR will be opened to merge to `main`.

**Pull request merged**